src/py42/sdk/__init__.py

```diff
@@ -134,25 +163,121 @@ def securitydata(self):
             * Security plan information
 
         Returns:
-            :class:`py42.modules.securitydata.SecurityModule`
+            :class:`py42.services.securitydata.SecurityDataClient`
         """
-        return self._sdk_dependencies.security_module
+        return self._clients.securitydata
 
     @property
     def detectionlists(self):
         """A collection of properties each containing methods for managing specific detection
         lists, such as departing employees.
 
         Returns:
-            :class:`py42.modules.detectionlists.DetectionListsModule`
+            :class:`py42.services.detectionlists.DetectionListsClient`
         """
-        return self._sdk_dependencies.detection_lists_module
+        return self._clients.detectionlists
 
     @property
     def alerts(self):
         """A collection of methods related to retrieving and updating alerts rules.
 
         Returns:
-            :class:`py42.modules.alertrules.AlertRulesModule`
+            :class:`py42.services.alertrules.AlertRulesClient`
         """
-        return self._sdk_dependencies.alerts_module
+        return self._clients.alerts
+
+
+def _init_services(main_connection, main_auth):
+    alert_rules_key = u"FedObserver-API_URL"
+    alerts_key = u"AlertService-API_URL"
+    file_events_key = u"FORENSIC_SEARCH-API_URL"
+    preservation_data_key = u"PRESERVATION-DATA-SERVICE_API-URL"
+    employee_case_mgmt_key = u"employeecasemanagement-API_URL"
+    kv_prefix = u"simple-key-value-store"
+
+    kv_connection = Connection.from_microservice_prefix(main_connection, kv_prefix)
+    kv_service = KeyValueStoreClient(kv_connection)
+
+    alert_rules_conn = Connection.from_microservice_key(
```

Comment:
note that we can create the microservice connections right at startup, the new implementation make it so that we don't look up the url of the microservice until the first time we try to make a request with the connection